hawkbit-dmf/hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/AmqpMessageHandlerService.java

```diff
@@ -212,7 +196,36 @@ private void registerTarget(final Message message, final String virtualHost) {
             final Target target = controllerManagement.findOrRegisterTargetIfItDoesNotExist(thingId, amqpUri);
             LOG.debug("Target {} reported online state.", thingId);
             sendUpdateCommandToTarget(target);
-        } catch (EntityAlreadyExistsException e) {
+        } catch (final EntityAlreadyExistsException e) {
+            throw new AmqpRejectAndDontRequeueException("Target already registered, message will be ignored!", e);
+        }
+    }
+
+    /**
+     * Method to register a new target.
+     *
+     * @param message
+     *            the message that contains the target/thing
+     * @param virtualHost
+     *            the virtualHost of the target/thing
+     * @param name
+     *            the name of the target/thing, can be null and would then be
+     *            replaced by targetId
+     */
+    private void registerTarget(final Message message, final String virtualHost, final String name) {
```

Comment:
You don't have to overload registerTarget(). 
You can extract the name property from the message inside registerTarget to differentiate between the findOrRegisterTargetIfItDoesNotExist calls.

Reply:
It s not overloaded anymore and handled in findOrRegisterTargetIfItDoesNotExist.